gammapy/datasets/metadata.py

```diff
@@ -0,0 +1,103 @@
+from typing import ClassVar, Literal, Optional, Union
+import numpy as np
+from astropy.coordinates import SkyCoord
+from pydantic import ConfigDict
+from gammapy.utils.metadata import (
+    METADATA_FITS_KEYS,
+    CreatorMetaData,
+    MetaData,
+    ObsInfoMetaData,
+    PointingInfoMetaData,
+)
+
+__all__ = ["MapDatasetMetaData"]
+
+MapDataset_METADATA_FITS_KEYS = {
+    "MapDataset": {
+        "creation": "CREATION",
+        "event_types": "EVT_TYPE",
+        "optional": "OPTIONAL",
+        "pointing": "POINTING",
```

Comment:
without adding `pointing`, `to_header` fails if `altaz` is not present

Reply:
See #5075 and #5076.